Firestore/core/src/firebase/firestore/nanopb/writer.h

```diff
@@ -74,7 +54,44 @@ class Writer {
   explicit Writer(const pb_ostream_t& stream) : stream_(stream) {
   }
 
-  pb_ostream_t stream_;
+  pb_ostream_t stream_{};
+};
+
+/**
+ * Creates a Writer that writes into a vector of bytes.
+ *
+ * This is roughly equivalent to the nanopb function pb_ostream_from_buffer(),
+ * except that ByteStringWriter manages the buffer.
+ */
+class ByteStringWriter : public Writer {
+ public:
+  ByteStringWriter();
+
+  nanopb::ByteString ToByteString() const;
+
+  /**
+   * Returns the vector backing this ByteStringWriter.
+   */
+  std::vector<uint8_t> ToVector();
+
+ private:
+  std::vector<uint8_t> buffer_;
+};
+
+/**
+ * Creates an Writer that writes into a std::string.
+ *
+ * This is roughly equivalent to the nanopb function pb_ostream_from_buffer(),
+ * except that StringWriter manages the string.
+ */
+class StringWriter : public Writer {
+ public:
+  StringWriter();
+
+  std::string ToString();
```

Comment:
As above, but more dangerous, since ToString() is often used in logging/debugging.

Reply:
Yep. Fixed.